### DISCUSSION TITLE
Copilot is Crashing

### DISCUSSION THREAD

**Discussion by sysrpl** [community/community#45394] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I paid for a subscription to copilot this month but am having problems with Copilot crashing or stopping. I am running Linux Mint 21.1 with Dotnet 6 and 7, VS Code, and C#. The problem happens regularly under the following conditions:

I have a solution with C# files. 
If I open VS Code with this solution the Copilot works by suggesting code.

* but *

If I switch to another C# source code file inside the solution then back to the original file copilot either crashes or stops working altogether.

The only way to get Copilot working again is to exit and restart VS Code. 

What is going on and what steps can I take to gather more information about this problem in order to forward it to the developers in charge of Copilot and VS Code integration?

**Comment by jge162** [comment]
Check the VS Code and Copilot extension logs. You can find the log files in the following location:
Linux: ~/.vscode/logs/

Check for any other extensions you have installed that may be causing conflicts with Copilot. Try disabling other extensions temporarily to see if the issue persists.

Other than that great feedback. Hopefully this can be patched.

**Comment by sysrpl** [comment]
@jge162 Thank you for the feedback. On my Linux system there exists a ~/.vscode/ folder, but it does not contain any log folder or log files. Is there some setting I need to turn on to enable detailed logging either for VS Code or Copilot?

Regarding turning off other extensions, I had tried that before posting this question. I removed most all extensions other than C# supprot and Roslynator. Just now I tried removing Roslynator and the problem still presents itself.

**Reply by jge162** [reply]
@sysrpl You can enable logging by adding the following line to your settings.json file in VS Code:

"telemetry.enableCrashReporter": true

This will enable telemetry and crash reporting for VS Code, which will generate log files in the appropriate location.

for the Copilot extension add the following line to your settings.json file:

"copilot.logLevel": "trace"

This will enable trace-level logging for Copilot, which will provide more detailed information in the logs.

This should help you debug the issue and potentially resolve it. Best 👍

**Comment by microhobby** [comment]
I'm having similar behavior (paid copilot), from the output panel of VS Code `GitHub Copilot` I got the following logs:

```logs
[ERROR] [default] [2023-01-31T12:52:52.034Z] GitHub Copilot could not connect to server. Extension activation failed: "Command failed: dotnet build /t:GetRemoteControlHostPath /<I hide my local path>.csproj
"
```

This behavior only occurs for .NET workspace projects, tested on different projects.
